packages/material-ui/src/Backdrop/Backdrop.js

```diff
@@ -7,7 +7,6 @@ import Fade from '../Fade';
 export const styles = {
   /* Styles applied to the root element. */
   root: {
-    zIndex: -1,
```

Comment:
I'm probably overly naive here. But wouldn't you want the backdrop to block the main document? Therefore putting it behind sounds wrong. Is this required to make it work with other components?

Reply:
Try the scroll body demo. You should see a problem with the scroll.

Reply:
https://deploy-preview-14717--material-ui.netlify.com/demos/dialogs/#scrolling-long-content ? I don't "see a problem". Could explain what's expected, what actually happens and why this should happen?

Reply:
You can't scroll when hovering the backdrop.

Reply:
![mui-scroll](https://user-images.githubusercontent.com/12292047/53716722-2166b700-3e56-11e9-807a-615954c806cb.gif)

Reply:
I can’t scroll whilst hovering the backdrop on a phone or desktop either.

Reply:
I'm currently looking through some dialog/modal implementations and haven't found the scroll=body behavior anywhere else. What was the original motivation for this behavior?

Looking into wai-aria authoring practices as well but I suspect it's considered best practice to put the scrollbar next to the content it scrolls. It may be irritating that a scrollbar the previously scrolled the document now scrolls different content.

Then we also have the issue that the material spec tells use to make only the dialog content scrollable. The actions should be visible on the screen. Enforcing this improves a11y of modals as well and makes them less distracting.

Reply:
> What was the original motivation for this behavior?

Why are we supporting this scroll=body variant:
- #7438
- ![capture d ecran 2019-03-04 a 10 25 24](https://user-images.githubusercontent.com/3165635/53723321-d81f6300-3e67-11e9-9719-b5e1423370ff.png)
- ![capture d ecran 2019-03-04 a 10 29 28](https://user-images.githubusercontent.com/3165635/53723624-6dbaf280-3e68-11e9-9f45-6026e694407d.png)


> Enforcing this improves a11y of modals as well and makes them less distracting.

Are you suggesting that we should remove this **non-default** option? I think that we should provide the tools people ask for, let them choose what best fit their need while keeping the most reasonable choice as a default: scroll=paper.

Reply:
Somebody telling you they "would like that" is in no way sufficient reason to support it. I would like the component to give me candies whenever I open it. Is it reasonable to support it? Probably not.

Again do you have some sources for this behavior? Some text explaining why you want to detach the scrollbar from the scrollable content? I suspect that these were all motivated by "bad modal" behavior. Dialogs are very dangerous UX practice so I'm skeptic about any use case that even furthers this problematic behavior by abusing modals for nested windows.

All these are better served with full screen dialogs instead of partial dialogs + detached scrollbar.

>  I think that we should provide the tools people ask for

Nope, we should provide the tools to build good ui/ux. Somebody asking without giving a coherent argument is not sufficient. Especially if it implies surprising behavior close to being a bug (here backdrop clicks not registering on the backdrop).

Reply:
oh and I have forgotten, jQuery UI do that too.

Reply:
> Nope, we should provide the tools to build good ui/ux.

Sorry, I have updated the comment replacing "ask for" with "need".

Reply:
jQuery ui doesn't have a backdrop: https://codesandbox.io/s/7wx2k0mmrq. The rest of the document isn't inert i.e. if I scroll both body and modal scroll. The argument doesn't apply.

Reply:
@oliviertassinari  I'm not aiming for some gotcha here. This is me legit asking where you've seen scrollbars that are detached from the content they scroll and if you have some resources that describe this UX pattern.

> Sorry, I have updated the comment replacing "ask for" with "need".

If they justify that need sure. It doesn't look that way though looking through the linked issues/prs.

Reply:
Please, let's move to a different topic, we should keep the scroll body. I don't want to get into the why, I doubt there is any A + B = C type of logical discussion we can get on the subjet. I would rather save time by benchmarking what the others are doing. It's [well accepted](https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html) in literature and [real world applications](https://trello.com/). There were smarter people than me that came to this conclusion

- It's the default behavior of Bootstrap, Ant Design & jQuery UI, it's not our default behavior. maybe it should?
- 100,000 views on https://stackoverflow.com/questions/10476632/how-to-scroll-the-page-when-a-modal-dialog-is-longer-than-the-screen
- [Do you use Trello?](https://github.com/yapplabs/ember-modal-dialog/issues/11)

Reply:
> +100,000 views on stackoverflow.com/questions/10476632/how-to-scroll-the-page-when-a-modal-dialog-is-longer-than-the-screen

Accepted answer implements `scroll=paper`

> Completely covering the background window also hides background movement that occurs on some mobile devices when scrolling content inside the dialog.

describes `scroll=paper` doesn't implement it though.

> real world applications

Trello does it, accepted.

google products don't as far as I can tell.

Backdrop click bug is still an issue though

Reply:
> google products don't as far as I can tell.

Yes. I know that Gmail doesn't do it & Keep doesn't do it. I haven't look deeper.

> Backdrop click bug is still an issue though

Yeah, it was the cost of supporting this feature. @joshwooding & I couldn't find a better solution 😢.